src/rt/object.rs

```diff
@@ -106,6 +107,15 @@ impl Object {
             _ => None,
         }
     }
+
+    pub(super) fn rwlock_mut(self, store: &mut Store) -> Option<&mut rwlock::State> {
+        assert_eq!(self.execution_id, store.execution_id);
+
+        match &mut store.entries[self.index] {
```

Comment:
Why is the `&mut` required here? I rarely sees a `&mut` for match statement, I wonder what it does.

Reply:
In this case I was following the other `*_mut` methods. `mutext` does the same thing so I copy-pasted it. I can look into it, but I think consistency (all of the `*_mut` methods look _exactly_ the same) is important here so we should update all methods -- or I'll need to figure out why `rwlock` is different from `mut`.

Reply:
* To answer your question I think we `&mut` here because we are returning an `Option<&mut rwlock::State>` in the `match`. If I try to get rid of the `&mut` here I get messages like this:
```rs
error[E0308]: mismatched types
   --> src/rt/object.rs:115:38
    |
115 |             Entry::RwLock(v) => Some(v),
    |                                      ^
    |                                      |
    |                                      expected mutable reference, found struct `rt::rwlock::State`
    |                                      help: consider mutably borrowing here: `&mut v`
    |
    = note: expected type `&mut rt::rwlock::State`
               found type `rt::rwlock::State`
```

but I can't return `Some(&mut v)` because we're not borrowing as mutable.